src/components/authentication/callback_handler.tsx

```diff
@@ -1,28 +1,26 @@
-import React, { useEffect, useState, useContext } from 'react';
+import React, { useContext, useState } from 'react';
 import { SessionContext } from '../../context/session_context';
 import { useLocation, Redirect } from 'react-router';
-import { ConfigContext } from '../../context/config_context';
-import { ApiV1AuthenticationRepository } from '../../repositories/authentication/implementations/api_v1_repository';
 
 export const CallbackHandler = () => {
   const query = new URLSearchParams(useLocation().search);
   const code: string | null = query.get('code');
-  const ctx = useContext(SessionContext);
-  const [loginSuccess, setLoginSuccess] = useState(false);
-  const configContext = useContext(ConfigContext);
+  const sessionContext = useContext(SessionContext);
+  const [isHandlingCallback, setIsHandlingCallback] = useState<boolean>(false);
+  if (!code) {
+    // TODO: handle case where code is not present
+    return null;
+  }
+  if (!isHandlingCallback) {
```

Comment:
As I look at this again, this code probably ought to be placed in a `useEffect`... I think this will create a race condition potentially